src/psyclone/configuration.py

```diff
@@ -802,10 +809,58 @@ def __init__(self, config, section):
             elif key == "access_mapping":
                 # Handled in the base class APISpecificConfig
                 pass
+            elif key == "grid-properties":
+                # Field properties have the format:
+                # go_grid_area_u: {0}%%grid%%area_u: array,
+                # First the name, then the Fortran code to access the property
+                # then the type ("array" or "scalar")
+                all_props = self.create_dict_from_string(section[key])
+                for grid_property in all_props:
+                    try:
+                        access, field_type = all_props[grid_property]\
+                                            .split(":")
+                    except ValueError:
+                        # Raised when the string does not contain exactly
+                        # two values separated by ":"
+                        error = "Invalid property \"{0}\" found with value " \
+                                "\"{1}\" in \"{2}\". It must have exactly " \
+                                "two ':'' separated values." \
+                                .format(grid_property,
+                                        all_props[grid_property],
+                                        config.filename)
+                        raise ConfigurationError(error)
+                    # Make sure to remove the spaces which the config
+                    # file might contain
+                    self._grid_properties[grid_property] = \
```

Comment:
I think this would be nicer as a named Tuple. That way we the code that accesses these properties will be more readable - i.e. it won't just be  '[0]' and '[1]'.

Reply:
I had no idea about named tuples ... they are GREAT, and I can immediately think of a few other places where we have tuples atm, which should be convert ... but not as part of this PR.

But the grid_properties here have been converted to a named tuple.